core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -156,6 +162,13 @@ abstract class InterBrokerSendThread(
   def wakeup(): Unit = networkClient.wakeup()
 }
 
+abstract class InterBrokerRequestManager(interBrokerSendThread: InterBrokerSendThread) {
```

Comment:
I wonder if we could just set the interBrokerSendThread in the .addRequestManager method.  This way, we don't need to pass interBrokerSendThread in all request manager classes (and wakeup doesn't make sense anyway until the request manager is added to the interBrokerSendThread).

Reply:
Sure -- that makes sense.